api/v1alpha1/shared_types.go

```diff
@@ -474,6 +474,27 @@ type BackendRef struct {
 	gwapiv1.BackendObjectReference `json:",inline"`
 }
 
+type BackendCluster struct {
+	// BackendRef references a Kubernetes object that represents the
+	// backend server to which the authorization request will be sent.
+	//
+	// Deprecated: Use BackendRefs instead.
+	BackendRef *gwapiv1.BackendObjectReference `json:"backendRef,omitempty"`
+
+	// BackendRefs references a Kubernetes object that represents the
+	// backend server to which the authorization request will be sent.
+	//
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	BackendRefs []BackendRef `json:"backendRefs,omitempty"`
+
+	// BackendSettings holds configuration for managing the connection
+	// to the backend.
+	//
+	// +optional
+	BackendSettings *ClusterSettings `json:"backendSettings,omitempty"`
```

Comment:
Prefer to add `BackendSettings` to `BackendRef`, and remove the `BackendCluster` intermediate layer.

Reply:
`BackendSettings` can't be added to `BackendRef` because a `BackendRef` represents a single endpoint in an Envoy Proxy cluster and all of the attributes in `BackendSettings` are relevant for the entire envoy proxy cluster.